includes/rules-engine.php

```diff
@@ -41,6 +41,9 @@ function badgeos_bp_trigger_event( $args = '' ) {
 	if ( empty( $user_ID ) && 'bp_core_activated_user' == current_filter() )
 		$user_ID = absint( $args );
 
+	if ( 'groups_join_specific_group' == current_filter() )
+		$user_ID = absint( $args[1] );
+
```

Comment:
These lines are only really needed for the `groups_membership_accepted` trigger. But they work for all `groups_join_specific_group` triggers as `$args[0]` is the same as `$user_ID`.